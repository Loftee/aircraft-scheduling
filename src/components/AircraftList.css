.aircraft-list-wrapper {
    width: 25%;
    padding: 15px;
    float: left;
}

.aircraft-list {
    width: 100%;
    list-style: none;
    padding: 0;
}

.aircraft-list-item {
    margin-bottom: 20px;
    padding: 10px;
    border: 1px solid #ccc;
    cursor: pointer;
}

.aircraft-list-item:hover,
.aircraft-list-item--selected {
    background: #eee;
    border-color: #999;
}

.aircraft-list-item--selected {
    border-style: dashed;
}