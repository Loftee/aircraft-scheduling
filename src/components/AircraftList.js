import React, {Component} from 'react';

import './AircraftList.css';

class AircraftList extends Component {
    render() {
        const {aircraftList} = this.props;
        return (
            <ul className={'aircraft-list'}>
                {
                    aircraftList.map((aircraft, index) => {
                        return (
                            <li key={index} className={'aircraft-list-item'}>
                                <h3 className={'aircraft-list-item__header'}>Aircraft Id: {aircraft.id}</h3>
                                <div className={'aircraft-list-item__data'}>
                                    <span>Aircraft type: {aircraft.type}</span>
                                    <span>Aircraft economy seats available: {aircraft.economySeats}</span>
                                    <span>Aircraft base: {aircraft.base}</span>
                                </div>
                            </li>
                        )
                    })
                }
            </ul>
        )
    }
}

export default AircraftList;