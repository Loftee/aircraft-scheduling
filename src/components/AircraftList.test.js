import React from 'react';
import {shallow} from 'enzyme';
import AircraftList from './AircraftList';

describe('Aircraft list', () => {
    it('renders correctly', () => {
        const props = {aircraftList: []};
        const component = shallow(<AircraftList {...props} />);
        expect(component).toMatchSnapshot();
    });

    describe('when rendered', () => {
        const props = {
            aircraftList:
                [{
                    "ident": "GABCD",
                    "type": "A320",
                    "economySeats": 186,
                    "base": "EGKK"
                }, {
                    "ident": "STUVW",
                    "type": "A320",
                    "economySeats": 150,
                    "base": "LFSB"
                }]
        };
        const component = shallow(<AircraftList {...props} />);

        it('fetches and renders an aircraft list', () => {
            expect(component.find('.aircraft-list-item').length).toEqual(2);
        });
    });
});