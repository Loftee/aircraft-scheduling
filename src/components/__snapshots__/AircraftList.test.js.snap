// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aircraft list renders correctly 1`] = `
<div
  className="aircraft-list-wrapper"
>
  <h2>
    Select an aircraft
  </h2>
  <ul
    className="aircraft-list"
  >
    <li
      className="aircraft-list-item"
      key="0"
      onClick={[Function]}
    >
      <h3
        className="aircraft-list-item__header"
      >
        Aircraft Id: 
        GABCD
      </h3>
      <div
        className="aircraft-list-item__data"
      >
        <div>
          Aircraft type: 
          A320
        </div>
        <div>
          Aircraft economy seats available: 
          186
        </div>
        <div>
          Aircraft base: 
          EGKK
        </div>
        <div>
          Utilisation: 
        </div>
      </div>
    </li>
  </ul>
</div>
`;
